kyo-prelude/shared/src/main/scala/kyo/Abort.scala

```diff
@@ -6,11 +6,36 @@ import kyo.Tag
 import kyo.kernel.Effect
 import scala.annotation.targetName
 
-/** The Abort effect allows for short-circuiting computations with failure values. This effect is used for handling errors and early
-  * termination scenarios in a functional manner and handle thrown exceptions.
+/** The Abort effect represents early termination of computations with typed failure values.
+  *
+  * Abort provides a functional approach to error handling, allowing you to explicitly model failures in your type signatures and handle
+  * them in a composable way. It serves as a more powerful alternative to exceptions that integrates with Kyo's effect system.
+  *
+  * A computation can terminate through Abort in three distinct ways:
+  *   - `Success[A]`: The computation completes successfully with a value of type A
+  *   - `Failure[E]`: An expected business/domain failure with a meaningful error value of type E
+  *   - `Panic(ex: Throwable)`: An unexpected exception, similar to unchecked exceptions
+  *
+  * Type unions allow expressing multiple possible error types: `Result < Abort[NetworkError | ValidationError]` indicates the operation
```

Comment:
```suggestion
  * Type unions allow expressing multiple possible error types: `Value < Abort[NetworkError | ValidationError]` indicates the operation
```

Might be good to avoid shadowing the 'Result' type.